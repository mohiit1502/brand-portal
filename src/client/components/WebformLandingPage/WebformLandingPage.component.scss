@import "../../styles/override/variables";

.c-WebformLandingPage {
    padding: 1.5rem 2.5rem;
    .btn {
        width: 150px;
        &.btn-primary {
            background: white;
            color: #006799;
            border: 1px solid #006799;
            opacity: 1;
        }
    }

    .content-header {
        font-size: 24px;
        font-weight: bold;
        line-height: 1.08;
        margin-top: 0.5rem;
        min-height: 50px;
        color: #0D6799;
    }
}
