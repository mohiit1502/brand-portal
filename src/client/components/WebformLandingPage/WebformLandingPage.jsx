import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import "./WebformLandingPage.component.scss";

const WebformLandingPage = props => {
  const [contentConfig, setContentConfig] = useState({});
  useEffect(() => {
    setContentConfig(props.configuration.content ? props.configuration.content : {});
}, [props.configuration]);

  const contentRenders = props.getContent && contentConfig && Object.keys(contentConfig).map(node => {
    return props.getContent(contentConfig, node);
  });
  return (
    <div className="c-WebformLandingPage">
      <div className="page-body">
          <div className="page-header h4 font-weight-bold">
            {
              props.configuration.header.text
            }
          </div>
          <div>{contentRenders}</div>
      </div>
    </div>
  );
};

WebformLandingPage.propTypes = {
  getContent: PropTypes.func,
  configuration: PropTypes.Object
};

export default WebformLandingPage;
