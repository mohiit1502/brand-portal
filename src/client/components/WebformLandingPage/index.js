import WebformLandingPage from './WebformLandingPage.jsx';

export default WebformLandingPage;
