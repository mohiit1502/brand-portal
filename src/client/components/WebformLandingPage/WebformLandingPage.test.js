/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import WebformLandingPage from "./WebformLandingPage";

describe("WebformLandingPage", () => {
   it("renders without error", () => {

   });
});
