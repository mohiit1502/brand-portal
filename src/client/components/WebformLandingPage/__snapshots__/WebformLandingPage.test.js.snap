// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebformLandingPage renders without error 1`] = `
<WebformLandingPage
  configuration={
    Object {
      "disclaimer": Object {
        "actionBtnText": "OK",
        "btnText": "View disclaimer",
        "classes": "d-inline-block anchor-styled mb-35rem",
        "modalDisclaimerText": "To make a claim of IP infringement, you will be required to declare, under penalty of perjury, that the information provided is correct and that you are the IP rights owner or an authorized agent. If the information you submit is inaccurate or incomplete, we may be unable to process or respond to your request. If you have questions about IP or whether your rights have been infringed, consider seeking legal advice.",
        "modalHeaderText": "Disclaimer",
      },
      "header": Object {
        "classes": "h4 font-weight-bold",
        "text": "Intellectual Property (IP) Claim Form",
      },
      "help": Object {
        "actionBtnText": "OK",
        "classes": "d-inline-block anchor-styled",
        "modalHeaderText": "Help",
        "modalHelpText": Object {
          "content": Object {
            "partial1": Object {
              "anchor1": Object {
                "href": "mailto:ipinvest@walmart.com",
                "text": "ipinvest@walmart.com",
              },
              "chunk1": "If you have any issues with filing a claim, please reach out to ",
              "chunk2": " for further assistance.",
            },
          },
        },
      },
      "subText": Object {
        "classes": "d-inline-block mr-3 mb-35rem",
        "text": "All of the following fields are required",
      },
      "titleClass": "ml-8",
    }
  }
  getContent={[MockFunction]}
>
  <div
    className="c-WebformLandingPage"
  >
    <div
      className="page-body"
    >
      <div
        className="page-header h4 font-weight-bold"
      >
        Intellectual Property (IP) Claim Form
      </div>
      <div />
    </div>
  </div>
</WebformLandingPage>
`;
